ooniprobe/AppDelegate.mm

```diff
@@ -143,6 +155,53 @@ -(void)application:(UIApplication *)application didReceiveLocalNotification:(UIL
     //TODO start your long running bg task here
 }
 
+- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url {
+    //NSLog(@"url recieved: %@", url);
+    //NSLog(@"query string: %@", [url query]);
+    
+    //creating parameters dict
+    NSDictionary *dict = [DictionaryUtility parseQueryString:[url query]];
+    NSDictionary *parameters = [DictionaryUtility getParametersFromDict:dict];
+    
+    NSString *minimum_version = [parameters objectForKey:@"mv"];
+    if (minimum_version != nil){
+        if ([minimum_version compare:[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"] options:NSNumericSearch] == NSOrderedDescending) {
+            //actualVersion is lower than the requiredVersion
+            UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"ooniprobe_outdate", nil) message:NSLocalizedString(@"ooniprobe_outdate_msg", nil) delegate:nil cancelButtonTitle:NSLocalizedString(@"Cancel", nil) otherButtonTitles:NSLocalizedString(@"ok", nil), nil];
+            alertView.tag = 2;
+            [alertView show];
+        }
+        else {
+            NSString *action = [url host];
+            if ([action isEqualToString:@"nettest"]){
+                //For now checking only test name
```

Comment:
Does this mean that we should _check_ for more parameters or that for now the code is limited in that it does not actually run the test but just makes sure that the test name is okay?

Reply:
We only have test name to check for now